### PR TITLE
Feature/trieuth/t01 show the total number of jobs

### PR DESCRIPTION
**Pull request by HoangTrieu2803**

### CODE REVIEW

**Comment by nqbinh13** [comment]
Please go through all file codes and add types for them

**Comment by nqbinh13** [comment]
image
Try to clear all warning

**Review by nqbinh13 — CHANGES_REQUESTED** [review]
please fix these code review

File: src/containers/Home/Jobs/index.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { any } from 'prop-types';
+import React, { useEffect, useState } from 'react'
+import axios from 'axios';
+export default function Jobs() {
+
+    const [jobs , setJobs] = useState([]);
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i have done

File: src/containers/Home/Jobs/index.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                </li>
+            )
+        })
+    }
+    const paginate = (pageNumber:any) => setCurrentPage(pageNumber)
+    const renderJob = () => {
+        return currentJob?.map((item: any) => {
+            return (
+                <li className='list-group-item d-flex'>
+                    <img style={{height:"50px"}} src='https://itviec.com/rails/active_storage/representations/proxy/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBLzRvREE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0c0d2b976b107ef51289f0361754cea78ab9313e/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCem9MWm05eWJXRjBTU0lJY0c1bkJqb0dSVlE2RTNKbGMybDZaVjloYm1SZmNHRmtXd2RwUm1sRyIsImV4cCI6bnVsbCwicHVyIjoidmFyaWF0aW9uIn19--80ef1c6b35a79f9e028daec95b88475f53428fd7/floware-logo.png' />
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i have changed

File: src/containers/Home/Jobs/index.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                        <p>{item.congNghe}</p>
+                    </div>
+                </li>
+            )
+        })
+    }
+    return (
+        <div className='container mt-5'>
+            <ul className='list-group mb-2' id='listJobs'>
+                {renderJob()}
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i have created new component for renderJob

File: src/containers/Home/Jobs/index.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { any } from 'prop-types';
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i have done

File: src/containers/Home/index.tsx [hunk 5]
```diff
@@ -1,17 +1,16 @@
 import logo from '../../logo.svg';
 import React from "react";
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/index.tsx:2]: i've done

File: src/containers/Home/Jobs/index.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import React, { useEffect, useState } from 'react'
+import axios from 'axios';
+export default function Jobs() {
+
+    const [jobs , setJobs] = useState([]);
+    useEffect(()=>{
+        axios.get(`https://6183cb0191d76c00172d1b6b.mockapi.io/api/products`)
+        .then(res=>{
+         const data = res.data;
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i've have fixed

File: src/containers/Home/Search/index.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        setCity(data);
+       })
+       .catch(error => console.log(error));
+    },[])
+    const renderJobs = () =>{
+            return (
+                <h2>{jobs?.length} IT Jobs For "Chất" Developers</h2>
+            )
+    }
+    const renderLocation = () =>{
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.tsx:25]: i've created new component for it

File: src/containers/components/Navbar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+
+export default function Navbar() {
+  return (
+    <div>
+        <ul className="nav bg-dark">
+  <li className="nav-item">
+    <a className="nav-link active" aria-current="page" href="#"><img style={{width:"50%"}} src='https://itviec.com/assets/logo-itviec-4492a2f2577a15a0a1d55444c21c0fa55810822b3b189fc689b450fb62ce0b5b.png'/></a>
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/components/Navbar/index.tsx]: i've changed to external css

File: src/containers/Type/city.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+interface City{
```
Comment (nqbinh13) [inline src/containers/Type/city.ts:1]: wrong syntax interface

File: src/containers/components/Navbar/index.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+import "./style.css"
+export default function Navbar() {
+  return (
+    <div>
+        <ul className="nav bg-dark">
+  <li className="nav-item">
+    <a className="nav-link active" href='/' aria-current="page"><img alt='logoITviec' style={{width:"50%"}} src='https://itviec.com/assets/logo-itviec-4492a2f2577a15a0a1d55444c21c0fa55810822b3b189fc689b450fb62ce0b5b.png'/></a>
+  </li>
+  <li className="nav-item">
+    <a className="nav-link" href='Alljobs'>All jobs</a>
```
Comment (nqbinh13) [inline src/containers/components/Navbar/index.tsx:11]: you can use `href='/#'` to prevent reload page

Comment (HoangTrieu2803) [reply src/containers/components/Navbar/index.tsx:11]: i have changed it

File: src/index.tsx [hunk 11]
```diff
@@ -7,15 +7,21 @@ import './index.scss';
 import App from './containers/App';
 import * as serviceWorker from './serviceWorker';
 
+// import {createStore,applyMiddleware} from "redux";
```
Comment (nqbinh13) [inline src/index.tsx:10]: remove comment lines

File: src/containers/Home/Search/index.tsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+       })
+       .catch(error => console.log(error));
+
+       axios.get(`${process.env.REACT_APP_API_URL}/qltt`)
+       .then(res=>{
+        setCity(res.data);
+       })
+       .catch(error => console.log(error));
+    },[])
+    const RenderLocation = ():JSX.Element =>{
```
Comment (nqbinh13) [inline src/containers/Home/Search/index.tsx:22]: This should use as HTML element no need to create a Component here

File: src/containers/components/Navbar/index.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import "./style.css"
+export default function Navbar() {
+  return (
+    <div>
+        <ul className="nav bg-dark">
+  <li className="nav-item">
+    <a className="nav-link active" href='/' aria-current="page"><img alt='logoITviec' style={{width:"50%"}} src='https://itviec.com/assets/logo-itviec-4492a2f2577a15a0a1d55444c21c0fa55810822b3b189fc689b450fb62ce0b5b.png'/></a>
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/components/Navbar/index.tsx:8]: i've done

File: src/index.tsx (hunk 11 above)
Comment (HoangTrieu2803) [reply src/index.tsx:10]: i've removed

File: src/containers/Home/Search/index.tsx (hunk 12 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.tsx:22]: i've changed

File: src/containers/Type/city.ts (hunk 9 above)
Comment (HoangTrieu2803) [reply src/containers/Type/city.ts:1]: i've fixed

File: src/containers/Home/Search/style.css
Comment (nqbinh13) [inline src/containers/Home/Search/style.css:1]: convert this to .scss file

File: src/containers/components/Navbar/index.tsx (hunk 10 above)
Comment (nqbinh13) [reply src/containers/components/Navbar/index.tsx:11]: apply with other link elements

File: src/containers/Home/Search/index.test.tsx [hunk 14]
```diff
@@ -0,0 +1,27 @@
+import {render,screen} from "@testing-library/react";
+import Search from "./index";
+import axios from "axios";
+test('renders inital document', () => {
+    render(<Search/>);
+    const divEle = screen.getByTestId("jobTest");
+    expect(divEle).toBeInTheDocument;
```
Comment (nqbinh13) [inline src/containers/Home/Search/index.test.tsx:7]: should `toBeInTheDocument()`

File: src/containers/Home/Search/index.test.tsx [hunk 15]
```diff
@@ -0,0 +1,27 @@
+import {render,screen} from "@testing-library/react";
+import Search from "./index";
+import axios from "axios";
+test('renders inital document', () => {
+    render(<Search/>);
+    const divEle = screen.getByTestId("jobTest");
```
Comment (nqbinh13) [inline src/containers/Home/Search/index.test.tsx:6]: - don't use `divEle`, should put name clear meaning, ex: `jobElement`
- And you should test all the Search bars, not only the `h2` element, maybe you can put the test-id in the bigger scope

File: src/containers/Home/Search/index.tsx [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+                        <h2 data-testid='jobTest' className='total-job'>{jobs?.length} IT Jobs For "Chất" Developers</h2>
+                        <form className="d-flex">
+                            <input value="Search" className="form-control me-2" type="search" placeholder="Search" aria-label="Search" />
+                            <button className="btn btn-danger" type="submit">Search</button>
+                        </form>
+                    </div>
+                </div>
+            </div>
+        </div>
+        // <>test</>
```
Comment (nqbinh13) [inline src/containers/Home/Search/index.tsx:44]: remove this

File: src/containers/Home/Search/index.test.tsx [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+    expect(divEle).toBeInTheDocument;   
+});
+test('render 100 jobs', async ()=>{
+    let data =0;
+    await axios.get(`${process.env.REACT_APP_API_URL}/products`)
```
Comment (nqbinh13) [inline src/containers/Home/Search/index.test.tsx:11]: should not call API here

File: src/containers/Home/Search/index.tsx [hunk 18]
```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState } from 'react'
+import "./style.css"
+import axios from 'axios';
+import City from '../../Type/city';
+import Job from '../../Type/job';
+function Search() {
+    const [jobs, setJob] = useState<Job[]>();
```
Comment (nqbinh13) [inline src/containers/Home/Search/index.tsx:7]: should `setJobs`

File: src/containers/Type/city.ts
Comment (nqbinh13) [inline src/containers/Type/city.ts:1]: if types city and job belong to the Search component, I think you should create `types.ts` file in the Search folder and define both of them into this

File: src/containers/Home/Search/style.css
Comment (HoangTrieu2803) [reply src/containers/Home/Search/style.css:1]: i've converted 
@nqbinh13

File: src/containers/Home/Search/index.test.tsx (hunk 15 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.test.tsx]: i've changed the name 
@nqbinh13

File: src/containers/Home/Search/index.test.tsx [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+       .then(res=>{
+        data = res.data.length;
+       })
+       .catch(error => console.log(error));
+    expect(data).toBe(100);
+})
+test('selection default is All Cities',  ()=>{
+    render(<Search/>)
+    const seclectEle = screen.getByDisplayValue("All Cities"); 
+    expect(seclectEle).toBeInTheDocument;
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.test.tsx]: i've changed 
@nqbinh13

File: src/containers/Home/Search/index.tsx (hunk 16 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.tsx]: i've removed 
@nqbinh13

File: src/containers/Home/Search/index.tsx (hunk 18 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.tsx]: i've changed
@nqbinh13

File: src/containers/Type/city.ts
Comment (HoangTrieu2803) [reply src/containers/Type/city.ts:1]: i've moved type into Search component
@nqbinh13

File: src/containers/components/Navbar/index.tsx (hunk 10 above)
Comment (HoangTrieu2803) [reply src/containers/components/Navbar/index.tsx]: i've changed

File: src/containers/Home/Search/index.test.tsx (hunk 17 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.test.tsx]: i've solved it

**Pull request merged**